internal/services/contracts_events_consumer.go

```diff
@@ -698,6 +711,136 @@ func (c *ContractsEventsConsumer) aftermarketDeviceAddressReset(e *ContractEvent
+func (c *ContractsEventsConsumer) vehicleNodeBurned(e *ContractEventData) error {
+	ctx := context.Background()
+	if e.ChainID != c.settings.DIMORegistryChainID || e.Contract != common.HexToAddress(c.settings.DIMORegistryAddr) {
+		return fmt.Errorf("vehicle burn from unexpected source %d/%s", e.ChainID, e.Contract)
+	}
+
+	var args contracts.RegistryVehicleNodeBurned
+	if err := json.Unmarshal(e.Arguments, &args); err != nil {
+		return err
+	}
+
+	c.log.Info().Int64("vehicleNode", args.VehicleNode.Int64()).Msg("burning vehicle node")
+	tx, err := c.db.DBS().Reader.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback() //nolint
+
+	ud, err := models.UserDevices(
+		models.UserDeviceWhere.TokenID.EQ(
+			types.NewNullDecimal(new(decimal.Big).SetBigMantScale(args.VehicleNode, 0))),
+		qm.Load(models.UserDeviceRels.BurnRequest),
+	).One(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	if _, err := ud.Delete(ctx, tx); err != nil {
+		return err
+	}
+
+	if mtr := ud.R.BurnRequest; mtr != nil {
+		mtr.Hash = null.BytesFrom(e.TransactionHash.Bytes())
+		mtr.Status = models.MetaTransactionRequestStatusConfirmed
+		if _, err := mtr.Update(ctx, tx, boil.Infer()); err != nil {
+			return err
+		}
+	}
+
+	c.log.Info().Int64("vehicleNode", args.VehicleNode.Int64()).Msg("vehicle node burned")
+	return tx.Commit()
+}
+
+func (c *ContractsEventsConsumer) vehicleNodeMintedWithDeviceDefinition(e *ContractEventData) error {
+	if e.ChainID != c.settings.DIMORegistryChainID || e.Contract != common.HexToAddress(c.settings.DIMORegistryAddr) {
+		return fmt.Errorf("vehicle mint from unexpected source %d/%s", e.ChainID, e.Contract)
+	}
+
+	ctx := context.Background()
+	var args contracts.RegistryVehicleNodeMintedWithDeviceDefinition
+	if err := json.Unmarshal(e.Arguments, &args); err != nil {
+		return fmt.Errorf("failed to unmarshal arguments from mint event: %w", err)
+	}
+
+	log := c.log.With().Int64("vehicleNode", args.VehicleId.Int64()).Int64("manufacturerId", args.ManufacturerId.Int64()).Str("deviceDefinitionId", args.DeviceDefinitionId).Logger()
+	log.Info().Msg("Minting vehicle with device definition")
+
+	tx, err := c.db.DBS().Writer.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to start transaction: %w", err)
+	}
+	defer tx.Rollback() //nolint
+
+	if check, err := models.MetaTransactionRequests(
+		models.MetaTransactionRequestWhere.Hash.EQ(null.BytesFrom(e.TransactionHash.Bytes())),
+	).Exists(ctx, tx); err != nil {
+		return fmt.Errorf("failed to check if vehicle mint event has already been handled: %w", err)
+	} else if check {
+		return fmt.Errorf("vehicle mint event has already been handled. tx hash: %s", e.TransactionHash.String())
+	}
+
+	userIDArgs := dex.IDTokenSubject{
+		UserId: args.Owner.Hex(),
+		ConnId: "web3",
+	}
+	userID, err := proto.Marshal(&userIDArgs)
+	if err != nil {
+		return fmt.Errorf("failed to marshal user id: %w", err)
+	}
```

Comment:
Shame on me for not doing this before—could we make like `func addressToUserID(common.Address) (string, error)`?